docs/content/topics/operators/deploy.md

```diff
@@ -11,4 +11,429 @@ aliases:
+
+```console
+$ kubectl -n kube-system get svc nginx-ingress-ingress-nginx-controller
+```
+
+It may take a few minutes for the address to be provisioned.  Once provisioned,
+capture the value under the `EXTERNAL-IP` header.
+
+Lastly, let's add our first bit of configuration to the Brigade's chart values
+file you've saved locally (`brigade2-values.yaml`).  Under the
+`apiserver.service` section, set `type` to `ClusterIP`.  Here is a condensed
+view with other nearby fields omitted:
+
+```yaml
+apiserver:
+  service:
+    type: ClusterIP
+```
+
+In doing so, Brigade's own API server is only reachable internally to the
+Kubernetes cluster.  All inbound traffic will be handled by the NGINX Ingress
+Controller.
+
+[chart's GitHub repository]: https://github.com/kubernetes/ingress-nginx/tree/main/charts/ingress-nginx
+[NGINX Ingress Controller]: https://kubernetes.github.io/ingress-nginx/
+
+### DNS hostname
+
+With the external IP address of the ingress controller in hand, you're now
+ready to create the DNS entry.  With your DNS provider of choice, you'll create
+an [A record] mapping the the hostname on a domain you control to this IP
+address.  For example, if you control the `example.com` domain and wish for the
+hostname of the Brigade server to be `mybrigade.example.com`, create the DNS
+record with `mybrigade` as the name, `A` as the record type and the
+external IP address captured previously as the entry in the corresponding IP
+addresses section.
+
+[A record]: https://www.cloudflare.com/learning/dns/dns-records/dns-a-record/
+
+### Adding Ingress configuration
+
+Now we're ready to add configuration for ingress to the chart values file for
+Brigade.  We'll also update this configuration after the next
+[Certificates and TLS](#certificates-and-tls) section.
+
+All configuration will go under the `apiserver` section of the
+`brigade-values2.yaml` that you've saved locally above.  Here's the breakdown,
+with other nearby sections omitted:
+
+```yaml
+apiserver:
+
+  ## This is the DNS hostname you've reserved
+  host: mybrigade.example.com
+
+  ## Ensure tls is enabled for all Brigade's internal components
+  ## For now, we'll have them use self-signed certs
+  tls:
+    enabled: true
+    generateSelfSignedCert: true
+    # cert: base 64 encoded cert goes here
+    # key: base 64 encoded key goes here
+
+  ## Here's where we configure ingress
+  ingress:
+    enabled: true
+    ## These are annotations specific to the ingress controller we chose,
+    ## the NGINX Ingress Controller
+    annotations:
+      kubernetes.io/ingress.class: nginx
+      nginx.ingress.kubernetes.io/secure-backends: "true"
+      nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+    tls:
+      enabled: true
+      ## We set this to false as we'll use an external certificate provider
+      ## in the next step
+      generateSelfSignedCert: false
+```
+
+Save these changes to `brigade-values2.yaml` and proceed to the next section.
+
+## Certificates and TLS
```

Comment:
I agree that TLS (with trusted certs) is a component of a production-grade deployment.

But are we going too far if we go into detail on deployment and use of a cert manager?

Should we consider narrowing the focus to cert-related option in the `values.yaml` (which includes options that don't involve a cert manager)?

Reply:
I suppose I agree that use of cert manager is a specific scenario.  I'd love to preserve some version of this demonstration of using cert manager (and specific ingress controller, etc.) as I found a similar doc useful for setting up Brigade v1 in a secure manner... what are your thoughts here?  Maybe save for a blog post?  Or another 'example setup' document?

Reply:
Both with and without ingress, we have provided four options for TLS:

1. Disable TLS
1. Let the chart generate a self-signed cert
1. Paste the cert and key straight into values.yaml
1. Do none of the above, and just make sure the cert secret exists through other means. This could be creating it yourself or allowing cert-manager or similar to create it.

imho, we should cover these options, but should stop short of becoming a cert-manager how to. imho, those who are inclined to use cert-manager probably don't need our hand holding.

Also, this is NOT a hill I am willing to die on. So if you disagree with me, that's fine. Please don't consider this a blocker.

> Maybe save for a blog post? Or another 'example setup' document?

I do want to start working soon on a series of short videos that cover specific topics. I think one on production-grade deployment would be perfect and we could quite reasonably cover this (to a certain extent) in that video. We can start with the prep work already done-- "We've deployed Nginx Ingress and Cert Manager to this cluster already and we're going to be integrating our production grade deployment with those services."

wdyt?

Reply:
Ok, agreed we should focus on discussion all configuration options in this doc rather than sticking to any one specific approach.  Will remove extended ingress controller and cert manager setup and replace with high-level listing of config opts...

Reply:
@krancour PTAL at https://github.com/brigadecore/brigade/pull/1518/commits/f7f7e67d6a51625c33e608075c53548b39702aea when convenient.

Reply:
👍